### PR TITLE
fix(iterable): batch job order

### PR DESCRIPTION
**Pull request by mihir-4116**
## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/iterable/util.js [hunk 1]
```diff
@@ -734,7 +733,43 @@ const filterEventsAndPrepareBatchRequests = (transformedEvents) => {
...
 };
 
+/**
+ * Groups events with the same message type together in batches.
+ * Each batch contains events that have the same message type and are from different users.
+ * 
+ * @param {Array} inputs - An array of events
+ * @returns {Array} - An array of batches
+ */
+const orderEvents = (inputs) => {
```
Comment (koladilip) [inline src/v0/destinations/iterable/util.js:743]: can you write unit tests for this?

Comment (mihir-4116) [reply src/v0/destinations/iterable/util.js:743]: added unit test cases

File: src/v0/destinations/iterable/util.js [hunk 2]
```diff
@@ -734,7 +733,43 @@ const filterEventsAndPrepareBatchRequests = (transformedEvents) => {
...
+      const currentMessageType = currentInput.message.type;
+      const currentUser = currentInput.metadata.userId;
+      if (currentMessageType === messageType && !userOrderTracker[currentUser]) {
+        batch.push(currentInput);
+      } else {
+        updatedInputsArray.push(currentInput);
+        userOrderTracker[currentUser] = true;
+      }
+    })
+    if (batch.length > 0) {
```
Comment (koladilip) [inline src/v0/destinations/iterable/util.js:762]: this condition is not required as there will at least one element in each batch: `const batch = [event];`

File: src/v0/destinations/iterable/util.js [hunk 3]
```diff
@@ -734,7 +733,43 @@ const filterEventsAndPrepareBatchRequests = (transformedEvents) => {
...
+ * Each batch contains events that have the same message type and are from different users.
+ * 
+ * @param {Array} inputs - An array of events
+ * @returns {Array} - An array of batches
+ */
+const orderEvents = (inputs) => {
+  const batches = [];
+  let currentInputsArray = inputs;
+  while (currentInputsArray.length > 0) {
+    const updatedInputsArray = [];
```
Comment (koladilip) [inline src/v0/destinations/iterable/util.js:747]: updatedInputsArray -> remainingInputsArray?

File: src/v0/destinations/iterable/util.js (hunk 1 above)
Comment (koladilip) [inline src/v0/destinations/iterable/util.js:743]: orderEvents  -> batchEvents?

Comment (mihir-4116) [reply src/v0/destinations/iterable/util.js]: updated

**Pull request merged**